extension/js/addTopicButtons.js

```diff
@@ -0,0 +1,265 @@
+}
+// function addQuoteLastButton(btn, isPosting) {
+
+    // let href = getQuoteURL(btn)
+    // if (!href){
+
+        // return;
+    // }
+    // var onclick = null;
+    // if (isPosting){
+       // onclick = "last" + btn.querySelector('i.icon.fa-quote-left.fa-fw').parentElement.getAttribute('onclick');
+// //       button.a.setAttribute("onclick", "last" + onclick);
+    // }
+    // let button = createButton(href + '&last=true', 'fa-quote-left last', 'ציטיר בלויז די לעצטע תגובה', 'ציטיר לעצטע', onclick, '1');
+
+    // btn.appendChild(button.li);
+// }
+
+
+function addDefaultPage(){
+    if (document.querySelector(`#kf-app-default-page`)){
+        return false
+    }
+    let li = document.createElement("li")
+    let a = document.createElement('a');
+    li.appendChild(a)
+    li.setAttribute("class", "rightside")
+    li.setAttribute("id", "kf-app-default-page")
+    a.setAttribute('onClick', "saveDefaultPage()")
+    a.innerText = "מאך די בלאט די דיפאולט בלאט"
+    let pagination = document.querySelectorAll("#nav-footer").item(0)
+    if (pagination){
+        pagination.insertBefore(li, pagination.firstChild);
+        return true;
+    }
+    return false;
+}
+
+function getPostLink(postID){
+    return `https://www.IVELT.com/forum/viewtopic.php?p=${postID}#p${postID}`;
+}
+
+function copyQuoteParse(post_id){
+    var html = document.querySelector(`#post_content${post_id} .content`).innerHTML
+    let post_url = getPostLink(post_id)
+    var username = getUsername(post_id)
+    let converter = new HTML2BBCode();
+    html = html.replaceAll("./download", "www.ivelt.com/forum/download")
+    navigator.clipboard.writeText(`[quote="${username}"]${converter.feed(html)} [/quote] [url=${post_url}]מקור[/url]`)
+}
+
+function getPostDetails(post_id, prefix = 'p'){
+    var usernameE = document.querySelector(`#${prefix}${post_id} .username`)
+    if (!usernameE){
+       var usernameE = document.querySelector(`#p${prefix}${post_id} .username-coloured`)
+    }
+    var username = ""
+    if(usernameE){
+       username = usernameE.innerText
+    }
+
+    var idE = document.querySelector(`#${prefix}${post_id} .username`);
+    if (!idE){
+       var usernameE = document.querySelector(`#p${prefix}${post_id} .username-coloured`)
+    }
+    var usernameLink = ""
+    let id =""
+    if(idE){
+       usernameLink = idE.href;
+       id = usernameLink.split("u=")[1];
+    }
+    var tsE = document.querySelector(`#${prefix}${post_id} time`)
+    if (!tsE){
+        var tsE = document.querySelector(`#${prefix}${post_id} [href='#postingbox']`).getAttribute('onclick')
+        if(tsE)
+            var time = tsE.match('(?<=time:)(.*)(?=,user)')[0];
+    }else{
+        var ts = Date.parse(tsE.dateTime)
+        var time = ts / 1000
+    }
+
+
+
+    return {
+        "username":username,
+        "id":id,
+        "time":time
+    };
```

Comment:
```suggestion
    let usernameE = document.querySelector(`#${prefix}${post_id} .username`);
    if (!usernameE){
       usernameE = document.querySelector(`#p${prefix}${post_id} .username-coloured`)
    }
    let username = "";
    if(usernameE){
       username = usernameE.innerText
    }

    const idE = document.querySelector(`#${prefix}${post_id} .username`);
    let usernameLink = "";
    let id =""
    if(idE){
       usernameLink = idE.href;
       id = usernameLink.split("u=")[1];
    }
    let tsE = document.querySelector(`#${prefix}${post_id} time`);
    let postTime;
    if (!tsE){
        tsE = document.querySelector(`#${prefix}${post_id} [href='#postingbox']`).getAttribute('onclick');
        if(tsE)
            postTime = tsE.match('(?<=time:)(.*)(?=,user)')[0];
    } else {
        const ts = Date.parse(tsE.dateTime);
        postTime = ts / 1000
    }

    return {
        "username":username,
        "id":id,
        "time":postTime
    };
```

there is a few multiple declarations and unused variables,